<?xml version="1"?>
<launch>
    <arg name="localization" default="false"/>
    <arg name="wait_for_transform" default="0.2"/>
    <arg name="database_path" default="rtabmap.db"/>

    <arg if="$(arg localization)" name="args" default=""/>
    <arg unless="$(arg localization)" name="args" default="--delete_db_on_start"/>

    <group ns="rtabmap">

        <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="$(arg args)">
            <param name="frame_id" type="string" value="base_link"/>

            <param name="subscribe_depth" type="bool" value="false"/>
            <param name="subscribe_rgb" type="bool" value="false"/>
            <param name="subscribe_rgbd" type="bool" value="true"/>
            <param name="subscribe_scan" type="bool" value="false"/>
            <!-- inputs -->
            <remap from="odom" to="/odom"/>
<!--            <remap from="scan" to="/scan"/>-->
            <remap from="rgbd_image" to="rgbd_image"/>
            <remap from="rgb/image" to="/camera/rgb/image_raw"/>
            <remap from="depth/image" to="/camera/depth/image_raw"/>
            <remap from="rgb/camera_info" to="/camera/rgb/image_converted/camera_info"/>
            <param name="queue_size" type="int" value="50"/>
        </node>
    </group>
</launch>
